Subject: Linux-RT 2.6.24-rc5-rt1
From: http://www.kernel.org/pub/linux/kernel/projects/rt/
Acked-by: Sven-Thorsten Dietrich <sdietrich@suse.de>
From: Sven-Thorsten Dietrich <sdietrich@novell.com>
Subject: KDB locking update for RT
Patch-Mainline: never

Convert kdb_lock to RAW_SPINLOCK. 

KDB is designed to operate non-preemptibly, and it deadlocks with 
itself when the kdb_lock operates as a mutex in the RT Kernel,

While there will be additional issues to resolve with KDB, this change 
is a step along the way to getting stack traces from KDB, that don't
pertain to internal KDB failures.

signed-off-by: Sven-Thorsten Dietrich <sdietrich@novell.com>

---
 kdb/kdbmain.c |    2 +-
 1 files changed, 1 insertion(+), 1 deletion(-)

--- linux-2.6.22.orig/kdb/kdbmain.c
+++ linux-2.6.22/kdb/kdbmain.c
@@ -58,7 +58,7 @@ atomic_t kdb_8250;
  * kdb_lock protects updates to kdb_initial_cpu.  Used to
  * single thread processors through the kernel debugger.
  */
-static DEFINE_SPINLOCK(kdb_lock);
+static DEFINE_RAW_SPINLOCK(kdb_lock);
 volatile int kdb_initial_cpu = -1;		/* cpu number that owns kdb */
 int kdb_seqno = 2;				/* how many times kdb has been entered */
 
